tests/filecheck/backend/riscv/func_and_arith_to_riscv_asm_flow.mlir

```diff
@@ -0,0 +1,31 @@
+// RUN: xdsl-opt %s -p convert-func-to-riscv-func,convert-snrt-to-riscv,convert-arith-to-riscv,reconcile-unrealized-casts,riscv-allocate-registers -t riscv-asm | filecheck %s
+
+
+func.func @test(%dst: i32, %src: i32) -> i32 {
+
+  %stride = arith.constant 128 : i32
+  %size = arith.constant 512 : i32
+  %repeat = arith.constant 64 : i32
+
+  %tx_id = "snrt.dma_start_2d"(%dst, %src, %stride, %stride, %size, %repeat) : (i32, i32, i32, i32, i32, i32) -> i32
+
+  func.return %tx_id : i32
+}
+
+
+// CHECK-NEXT: .text
+// CHECK-NEXT: .globl test
+// CHECK-NEXT: .p2align 2
+// CHECK-NEXT: test:
+// CHECK-NEXT:     mv t2, a0
+// CHECK-NEXT:     mv t3, a1
+// CHECK-NEXT:     li t1, 128
+// CHECK-NEXT:     li t0, 512
+// CHECK-NEXT:     li t4, 64
+// CHECK-NEXT:     dmsrc t3, zero
+// CHECK-NEXT:     dmdst t2, zero
+// CHECK-NEXT:     dmstr t1, t1
+// CHECK-NEXT:     dmrep t0
+// CHECK-NEXT:     dmcpyi t0, t0, 2
```

Comment:
Do these compile with a standard RISC-V compiler?

Reply:
Can we print these as `.word` s or `.insn` instead?

Reply:
I was under the impression that we want to do that separately, to have a way to print insns for certain instructions as an option?

Reply:
Yes, I have some big plans for the risc-v backend right now:
- Better infra to handle target printing (e.g. printing `.insn ...` things for instructions unsupported by our backend)
- A pass `convert-riscv-to-llvm` which wraps riscv instructions inside llvm inline assembly ops. For this we'll need to infer sideffects, which will be fun :D